File: mmoarpg-j21/cfg-client/code/Org/Wxd/Mmo/Core/Cfg/factory/QItemshopVipFactory.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Org.Wxd.Mmo.Core.Cfg.Bean;

namespace Org.Wxd.Mmo.Core.Cfg.Factory
{

    /// <summary>
    /// excel 构建 Vip礼包.xlsx - q_itemshop_vip - vip礼包
    /// <summary>
    public class QItemshopVipFactory
    {
        public QItemshopVipFactory()
        {
            string jsonString = File.ReadAllText("config_json/q_itemshop_vip.json");
            Beans = Newtonsoft.Json.JsonConvert.DeserializeObject<List<QItemshopVipBean>>(jsonString);
            foreach (var bean in Beans)
            {
                beanMap[bean.QId] = bean;
            }
        }

        public List<QItemshopVipBean> Beans { get; set; }
        
        private Dictionary<int, QItemshopVipBean> beanMap = new Dictionary<int, QItemshopVipBean>();

        public Dictionary<int, QItemshopVipBean> BeanMap
        {
            get { return beanMap; }
            set { beanMap = value; }
        }

    }
}

File: mmoarpg-j21/org.wxd.mmo.core/src/main/java/org/wxd/mmo/core/cfg/factory/QItemshopVipFactory.java
package org.wxd.mmo.core.cfg.factory;


import lombok.Getter;
import org.wxd.boot.batis.struct.DbBean;
import org.wxd.mmo.core.cfg.bean.QItemshopVipBean;

import java.io.Serializable;


/**
 * excel 构建 Vip礼包.xlsx - q_itemshop_vip - vip礼包
 *
 * @author: Troy.Chen(無心道, 15388152619)
 * @version: 2024-03-02 14:58:16
 **/
@Getter
public class QItemshopVipFactory extends DbBean<QItemshopVipBean> implements Serializable {

    @Override public void initDb() {
        /*todo 实现一些数据分组*/

    }

}